### ISSUE TITLE
Incorrect print on Linux Debian Stretch 4.9.13

### ISSUE THREAD

**Issue by androidrobotfactory** [alexrj/Slic3r#3828]
[config.ini.zip](https://github.com/alexrj/Slic3r/files/888027/config.ini.zip)

![screenshot](https://cloud.githubusercontent.com/assets/4927379/24575690/f91d896c-16f8-11e7-90f1-de0fb7183cd9.png)
[rip_5.zip](https://github.com/alexrj/Slic3r/files/887395/rip_5.zip)

### Version
1.2.9 and 1.2.9-780-g4c14c656

### Operating system type + version
Linux host 4.9.0-2-amd64 #1 SMP Debian 4.9.13-1 (2017-02-27) x86_64 GNU/Linux

### Behavior
* Bridge incorrectly added between gear cogs, see images. The one on the top with default print settings. Note the preview window does not show the problem. Same shape seen GCodeSimulator.jar so does not seem to be a printer problem. Thanks for taking a look. Added a simplified .stl with just two gears and the gcode. Conig file attached as config.ini.zip  (File → Export Config...).
* Appologies no related to OS version, worked OK when switched back to default print settings. Still trying to work out which print setting is causing the problem. Thanks for the assistance.

retract_before_travel = 2,2
use_firmware_retraction = 0

![gcode simulator](https://cloud.githubusercontent.com/assets/4927379/24575730/f191f704-16f9-11e7-855c-283e365574e3.png)

* Select 
[rip_4.stl.zip](https://github.com/alexrj/Slic3r/files/887124/rip_4.stl.zip)
Sailfish (Makerbot) as G-code flavor, open up rip_4.slt
![img_20170401_095320](https://cloud.githubusercontent.com/assets/4927379/24572290/1a11486a-16c3-11e7-8bad-66824368b566.jpg
)
![screenshot](https://cloud.githubusercontent.com/assets/4927379/24572579/82d00100-16c5-11e7-992b-074cd8c88abb.png)

#### STL/Config (.ZIP) where problem occurs
_Upload a zipped copy of an STL and your config (`File -> Export Config`)_

[rip_openscad.zip](https://github.com/alexrj/Slic3r/files/887126/rip_openscad.zip)

**Comment by lordofhyphens** [comment]
... what the heck? *sigh*

Stretch is -testing, isn't it? 

Is the preview panel showing anything re:bridge segments in Stretch?

**Comment by androidrobotfactory** [comment]
correct Stretch = Debian testing. That screen shot is from Stretch, no bridge. Only shows up when you open the generated gcode in the simulator or on the printer.

**Comment by xoan** [comment]
Please, upload your config (`File → Export Config...`).

It seems to be retraction related, probably `Minimun travel after retraction` parameter.

**Comment by lordofhyphens** [comment]
![image](https://cloud.githubusercontent.com/assets/31754/25114414/185fdda0-23ee-11e7-832d-6818a82166c0.png)

I can't seem to reproduce at all, the image here is from gcode.ws; I just updated this system to Stretch.

The travel moves also don't seem suspect at all. @androidrobotfactory Can you retest with a more recent checkout+build?

**Comment by alexrj** [comment]
@androidrobotfactory, the G-code file you attached doesn't show any issue. Moved between teeth are just travel moves (no extrusion). I opened it with multiple preview applications, and even your black screenshot seems to show just travel moves (white).

I think you just need to tune your retraction settings because probably retraction isn't happening. Maybe you want to lower your retraction speed.

I'm closing this issue since the uploaded files do not show any evidence of an issue. If you think we should investigate further, free to reopen including a G-code file which shows the problem.
